session.go

```diff
@@ -40,6 +42,7 @@ type session struct {
 type Config struct {
 	Keys          SessionKeys
 	Profile       ProtectionProfile
+	BufferFactory func(packetType packetio.BufferPacketType, ssrc uint32) packetio.Buffer
```

Comment:
BufferFactory should return an io.ReadWriteCloser.  The calling code should dynamically check whether the relevant methods are available, and ignore them if they do not.

Reply:
For RTCP/RTP paths is safe to stick with io.ReadWriteCloser, since is believe is the only useful case for the buffer factory I agree to make this change

Reply:
Done!